scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -403,12 +403,37 @@ class FormatOps(
   def templateCurly(template: Template): Option[Token] =
     getStartOfTemplateBody(template).map(x => nonCommentBefore(x).left)
 
-  def templateDerivesOrCurly(template: Template): Option[Token] = {
-    val endOfInits = template.inits.lastOption.flatMap(_.tokens.lastOption)
-    endOfInits.fold(templateCurly(template)) { x =>
-      if (x.end >= template.pos.end) None
-      else Some(nextNonComment(tokens(x)).left)
+  def templateDerivesOrCurly(
+      template: Template
+  )(implicit ft: FormatToken): Option[Token] =
+    findTemplateGroupOnRight { case (_, _, func) => func() }(template)
+      .orElse(templateCurly(template))
+
+  private def findTreeInGroup[A](
+      trees: Seq[Tree],
+      func: (Tree, Seq[Tree], () => T) => A
+  )(expireFunc: Seq[Tree] => T)(implicit ft: FormatToken): Option[A] =
+    trees.find(_.pos.end >= ft.right.end).map { x =>
+      func(x, trees, () => expireFunc(trees))
     }
+
+  def findTemplateGroupOnRight[A](
+      func: (Tree, Seq[Tree], () => T) => A
```

Comment:
```suggestion
      func: (Tree, Seq[Tree], ExpireFunction) => A
```

Could we add maybe a type alias here? Or even:

```suggestion
      func: (Tree, Group, ExpireFunction) => A
```

It was a bit hard to understand this function. `T` looked to be a type parameter and it not clear what this function would be used for.

Reply:
good idea, will do. thanks for the review.

btw, this is the last review, i don't have anything else pending.

Reply:
Ok, cool! Let's release RC2 to gather some more feedback and I can work on the rewrite to braces.

Reply:
on rc2, let's.

on "to braces": do we really need to? do you think it would be a common use case? https://dotty.epfl.ch/docs/reference/other-new-features/indentation.html#settings-and-rewrites

Reply:
adding back braces with multiple statements shouldn't be too hard. however, when it's one statement, we run into the same issue (mostly around infix) which has plagued #2019 and the issues it's superseded.

Reply:
Hmm... we could hold off for the time being and just add it if people indeed want it. But I guess in most cases people should just add `-no-indent`